great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -280,146 +125,63 @@ def add_year_and_month_splitter(
-            batch_requests.append(batch_request)
-        else:
-            # All options are defined by the splitter, so we look at its default values to fill
-            # in the option values.
-            default_option_values = []
-            for option in unspecified_options:
-                default_option_values.append(
-                    self.column_splitter.param_defaults(self)[option]
-                )
-            for option_values in itertools.product(*default_option_values):
-                # Add options from specified options
-                options = {
-                    name: batch_request.options[name] for name in specified_options
-                }
-                # Add options from unspecified options
-                for i, option_value in enumerate(option_values):
-                    options[unspecified_options[i]] = option_value
-                batch_requests.append(
-                    BatchRequest(
-                        datasource_name=batch_request.datasource_name,
-                        data_asset_name=batch_request.data_asset_name,
-                        options=options,
-                    )
-                )
-        return batch_requests
-
-    def _sort_batches(self, batch_list: List[Batch]) -> None:
-        """Sorts batch_list in place.
-
-        Args:
-            batch_list: The list of batches to sort in place.
-        """
-        for sorter in reversed(self.order_by):
-            try:
-                batch_list.sort(
-                    key=lambda b: b.metadata[sorter.metadata_key],
-                    reverse=sorter.reverse,
-                )
-            except KeyError as e:
-                raise KeyError(
-                    f"Trying to sort {self.name} table asset batches on key {sorter.metadata_key} "
-                    "which isn't available on all batches."
-                ) from e
-
-    def get_batch_list_from_batch_request(
-        self, batch_request: BatchRequest
-    ) -> List[Batch]:
-        """A list of batches that match the BatchRequest.
+    def _batch_from_fully_specified_batch_request(self, request: BatchRequest):
+        """Returns a Batch object from a fully specified batch request
 
         Args:
-            batch_request: A batch request for this asset. Usually obtained by calling
-                get_batch_request on the asset.
+            request: A fully specified batch request. A fully specified batch request is
+              one where all the arguments are explicitly set so corresponds to exactly 1 batch.
 
         Returns:
-            A list of batches that match the options specified in the batch request.
+            A single Batch object specified by the batch request
         """
-        # We translate the batch_request into a BatchSpec to hook into GX core.
-        self._validate_batch_request(batch_request)
-
-        batch_list: List[Batch] = []
+        batch_metadata = copy.deepcopy(request.options)
```

Comment:
This was the functionality that previously lived in the `for` loop of `get_batch_list_from_batch_request` and is unchanged.